### PR TITLE
Prony or Hankel method to extract multiple energy levels from a single correlator

### PR DESCRIPTION
**Pull request by urbach**
this is the Prony method which I have tried to provide now in a usable version.

### CODE REVIEW

File: R/hankel.R [hunk 1]
```diff
@@ -102,65 +111,139 @@ gevp.hankel <- function(cf, t0=1, deltat=1, n, N, eps=0.0001, range=c(0,1),
...
-bootstrap.hankel <- function(cf, t0, deltat=1, n, N, id=c(1), range=c(0,1), eps=0.001,
-                             element.order=c(1,2,3,4), submatrix.size=1) {
-  if(t0+2*n+deltat >= N) {
-    stop("t0+2*n+1+deltat must be smaller than N\n")
-  }
+bootstrap.hankel <- function(cf, t0, n=2, N, id=c(1), range=c(0,1), eps=0.001) {
+
+  stopifnot(inherits(cf, 'cf_meta'))
+  stopifnot(inherits(cf, 'cf_boot'))
+  if(missing(N)) N <- cf$Time/2+1
```
Comment (kostrzewa) [inline R/hankel.R]: I guess this should fail if the `cf` is not symmetrised, right?

**Comment by urbach** [comment]
> kostrzewa commented on this pull request.
> 
> 
> 
> >  #' @family hankel
> -bootstrap.hankel <- function(cf, t0, deltat=1, n, N, id=c(1), range=c(0,1), eps=0.001,
> -                             element.order=c(1,2,3,4), submatrix.size=1) {
> -  if(t0+2*n+deltat >= N) {
> -    stop("t0+2*n+1+deltat must be smaller than N\n")
> -  }
> +bootstrap.hankel <- function(cf, t0, n=2, N, id=c(1), range=c(0,1), eps=0.001) {
> +
> +  stopifnot(inherits(cf, 'cf_meta'))
> +  stopifnot(inherits(cf, 'cf_boot'))
> +  if(missing(N)) N <- cf$Time/2+1 
> 
> I guess this should fail if the `cf` is not symmetrised, right?

not neccessarily, it would still work, wouldn't it?

**Comment by j-ostmeyer** [comment]
The method is not applicable at all for `cosh`-type correlators, i.e. back-propagating parts destroy it. Correct?

**Comment by urbach** [comment]
Yes,  I think this is correct. But the idea is, of course, that this works in regions of time, where the back propagating part is not important.
Carsten Urbach, www.carsten-urbach.eu

**Comment by martin-ueding** [comment]
There is a build error and I cannot fix it in your repository, so I will open a new pull request.

**Pull request closed without merging**